### ISSUE TITLE
reference to an other entity

### ISSUE THREAD

**Issue by achecinski** [sebelga/gstore-node#120]
Hi, I have been looking for a way to add field in a schema that would be a reference to another entity.
Looking at the documentation I have not found anything that would allow to do it.
I'd like to do something similar to "Schema.Types.ObjectId" which exists in moongoose.
Does something like this already exist in GStore?
If not what would be the best way to store the key in the entity and load the referenced data when loading the referencing entity?

Thank you for your help.

**Comment by sebelga** [comment]
Hi,
For now there is no direct way to do this. But I think it would be a nice feature to add. You say that in mongoose, with the ObjectId type, it fetches all the _linked_ entities with 1 get call?

For now you could have a custom method to load the linked entity, and save the linked entity ID.

```js
// Example to load a linked Image entity for a User entity

const userSchema = new Schema( .... );

// Add a custom method on your Schema
userSchema.methods.getProfileImage = function getProfileImage() {
    return this.model('Image').get(this.imageId);
};

return gstore.model('User', userSchema);

// --------------

const User = require('./user.model');

async function fetchUser(id) {
    let userEntity;
    let user;
    try {
        userEntity = await User.get(id);

        // convert entity to plain object
        user = userEntity.plain();

        // add linked image entity to user object
        user.profilePict = await userEntity.getProfileImage();
    } catch(e) {
        // ...error handling
    }

    return user;
}
```

**Comment by achecinski** [comment]
Hi,

Thanks for your answer, I will try your code.
Actually there are two step in moongose to populate a foreign entity while populating the referencing entity.
First you need to set the foreign key type as Schema.Types.ObjectId and the you call the method populate() when retrieveing the referer : https://mongoosejs.com/docs/populate.html#population

**Comment by sebelga** [comment]
Hi,
Thanks for the link. I put this as a next feature to have in gstore-node. 
Cheers 👍

**Comment by sebelga** [comment]
Hi @achecinski 
I finally found the time to work on this feature. It has just been released in `v5.0.0`. Let me know your thoughts. Hopefully it will make your app logic much simpler!

Here are the docs: https://sebloix.gitbook.io/gstore-node/populate
